File: Webpage/server/webpage.server/webpage.server/src/main/java/abbruzzese/webpage/webpage/server/webpage/entities/EventImage.java
package abbruzzese.webpage.webpage.server.webpage.entities;

import java.util.UUID;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;

@Entity
public class EventImage {

    @Id
    private String id=UUID.randomUUID().toString();

    @Column(name="Estado",nullable = false)
    private String state;

    @Column(name="EventImage",nullable = false)
    private String eventImage;

    public EventImage(){}

    public EventImage(String State,String eventImage){

            this.state=State;
            this.eventImage=eventImage;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getEventImage() {
        return eventImage;
    }

    public void setEventImage(String eventImage) {
        this.eventImage = eventImage;
    }

}

File: PHP_API/ENTITIES/EventImage.php
<?php

namespace App\Entity;

use Ramsey\Uuid\Uuid;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\EventImageRepository;

#[ORM\Entity(repositoryClass: EventImageRepository::class)] // Esto usa atributos de PHP 8
#[ORM\Table(name: "eventimage")]
class EventImage{

    #[ORM\Id]
    #[ORM\Column(type:"string")]
    #[ORM\GeneratedValue(strategy:"NONE")]
    private $id;

    #[ORM\Column(type:"string")]
    private $estado;

    #[ORM\Column(type:"string")]
    private $eventimage;

    public function __construct($estado, $eventimage) {
       $this->id=Uuid::uuid4()->toString();
       $this->estado=$estado;
       $this->eventimage=$eventimage; 
    }

    //Getters y Setters

    public function getId() {return $this->id;}

    public function getEstado() {return $this->estado;}
    public function setEstado($estado){$this->estado;}

    public function getEventImage() {return $this->eventimage;}
    public function setEventImage($eventimage){$this->eventimage=$eventimage;}

}

?>